pkg/kubectl/cmd/patch.go

```diff
@@ -139,18 +139,17 @@ func RunPatch(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 	}
 
 	helper := resource.NewHelper(client, mapping)
-	_, err = helper.Patch(namespace, name, patchType, patchBytes)
+	patchedObject, err := helper.Patch(namespace, name, patchType, patchBytes)
 	if err != nil {
 		return err
 	}
 	if cmdutil.ShouldRecord(cmd, info) {
-		patchBytes, err = cmdutil.ChangeResourcePatch(info, f.Command())
-		if err != nil {
-			return err
-		}
-		_, err = helper.Patch(namespace, name, api.StrategicMergePatchType, patchBytes)
-		if err != nil {
-			return err
+		if err := cmdutil.RecordChangeCause(patchedObject, f.Command()); err == nil {
```

Comment:
This isn't in a loop or a separate method to do a `return` in.  I think this is the tightest representation.